unify page- and templateview

flx-sta opened the pull request:
Add a new component `<explorerpreviewcontent>` which is an abstract representation of pageview and templateview.


fix for issue #194

Commit:
METDEV-1654 change justify-content to space-between for explorer-confirm

- full selector: .explorer .explorer-layout .row .explorer-preview .explorer-preview-content .explorer-confirm-dialog

(cherry picked from commit 642f895cd2ab582e0c3db4c97d617db553db1314)

Commit:
METDEV-1654 change justify-content to space-between for explorer-confirm

- full selector: .explorer .explorer-layout .row .explorer-preview .explorer-preview-content .explorer-confirm-dialog

(cherry picked from commit 642f895cd2ab582e0c3db4c97d617db553db1314)

Commit:
METDEV-1654 revert justify-content back to flex-end

(cherry picked from commit 33c80a7062ae49622910e547bf50586260e5555f)

Commit:
METDEV-1654 revert justify-content back to flex-end

(cherry picked from commit 33c80a7062ae49622910e547bf50586260e5555f)

Commit:
METDEV-1654 add state action names (no new methods)

- deleteTemplate
- showTemplateInfo
- saveTemplateProperties

(cherry picked from commit a9b5819f8b87c8487bad397d8dfbb06025e5d52a)

Commit:
Merge branch 'develop' into metaproc-develop

* develop: (127 commits)
  added ability to direcly edit files with composum in the page tree
  Feature/sitedeletion (#179)
  added composum under tools
  Added function to create a package of all default site content when new sites are created. (#185)
  added path to output of tags and metaproperties
  another packagelock
  added package lock
  Updated Docker build script to exit the build if the Maven fails.
  Revert "Merge branch 'develop-sling11-merge' into develop"
  Added logic to site copy/creation process to update strings in files after copy. (#178)
  action now supports a visibility expression (#174)
  issues/175 - Updated package install script to handle installation of any new Peregrine packages.
  issues/175 - Initial work to discover and deploy all Peregrine packages during the Docker build.
  Rebuilt Peregrine Sling instance.
  improve drop target styles (#167)
  Updated Docker build to remove generated artifacts from files directory.
  fix for i18n issue causing all english terms to be lower cased
  switch from select to material-select for the metadata type
  combining meta name and properties into one field in the dialog
  updating the title for meta properties content
  ...

# Conflicts:
#	pagerenderer/vue/ui.apps/src/main/content/jcr_root/apps/pagerender/vue/components/placeholder/template.vue
#	pagerenderer/vue/ui.apps/src/main/content/jcr_root/apps/pagerender/vue/structure/page/explorer_dialog.json

NiklasFLuis committed:
Add og-tag functionality to componentDefinition

Add a functionality to retrieve and send the og-tag dialog-values with
the componentDefinitionServlet

Add constant OG_TAGS to PerConstants

NiklasFLuis committed:
Add tabs and og-tag dialog to pageview

Add functionality for different tabs and seperated dialogs to pageview
and add og-tag-tab and dialog

NiklasFLuis committed:
Add og-tags to populateComponentDefinitionFromNode

Add og-tags to populateComponentdefinitionFromNode to accept the new
content passed by the ComponentDefinitionServlet

Fix savePageProperties not working correctly due to change in the
object-structure and adding

NiklasFLuis committed:
Add og_tag_dialog.json with og-tags as content

Add dialog-file and content for og:title, og:description, og:type,
og:image, og:image:alt and og:url.

Add useTitle and useDescription dialog to allow for using the page title
and page description for og-tags

NiklasFLuis committed:
Add default values and meta-tags

Add default values to pagecontent.cnd for all available og-tags

Add meta-tags to page.html by using sly-test so tags only get rendered
if they are set

NiklasFLuis committed:
Copy pageview to templateview for consistency

Copy pageview structure to templateview for consistency and to implement
tab-functionality

NiklasFLuis committed:
Fix problems with new object structure

Fix problems in assetview, editor and objectview caused by the new
object structure of the ComponentDefinition response

NiklasFLuis committed:
Remove unneeded import

NiklasFLuis committed:
Fix ComponentDefinition returning model

Fix ComponentDefinition in combination with
populateComponentDefinitionFromNode directly returning the model instead
of an object including model an ogTags

NiklasFLuis committed:
Fix sly-tests in page

Fix sly-tests for meta-tags checking the wrong cases.

NiklasFLuis committed:
Change dialog descriptions

Change dialog descriptions to english and change the placeholder to the
og-tag. Eg. 'og:type' as placeholder for og:type

Commit:
METDEV-1654 add missing stateActions. improve explorerpreviewcontent template

(cherry picked from commit db7e90f859cc9bed9891aa2f490f20e4b5a63bba)

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/templateview/template.vue

Commit:
METDEV-1654 remove accidentally committed debugging code from pathbrowser

Commit:
METDEV-1654 add moveTemplate to api.js/apiImpl.js

Commit:
METDEV-1654 fix moveTemplate state-action

Commit:
METDEV-1654 remove debugging code from pathbrowser

Commit:
Revert "METDEV-1654 remove accidentally committed debugging code from pathbrowser"

This reverts commit f112d016

Commit:
METDEV-1654 fix falsely replaced pathBrowser string

Commit:
METDEV-1654 apply new explorerpreviewcontent component on pageview

- fix problems with "pages" and "sites"

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/pageview/template.vue

Commit:
remove og-tag feature from this branch

Commit:
remove obsolete v-else-if in the tabs

Commit:
remove another obsolete v-else-if in the tabs

Commit:
METDEV-1662 make og:image to browserType: image

(cherry picked from commit 1da79e69d634b00a023a7c37e21df7f5b709c6da)

Commit:
METDEV-1662 fix og-tag placeholders

(cherry picked from commit 22b2cf62b25b0d7b37d47a34f7ec028b5cf47f34)

# Conflicts:
#	pagerenderer/vue/ui.apps/src/main/content/jcr_root/apps/pagerender/vue/structure/page/og_tag_dialog.json

reusr1 reviewed (dismissed):
@Felix-Puetz it seems your change alters the style a bit and is not in line with the objects and assets preview (borders around the actions missing due to changing from li to a tag). I am also not sure about the icon on the left (it may be a good idea, however, then the assets and objects view need the same UX for consistency)

flx-sta commented:
> @Felix-Puetz it seems your change alters the style a bit and is not in line with the objects and assets preview (borders around the actions missing due to changing from li to a tag). I am also not sure about the icon on the left (it may be a good idea, however, then the assets and objects view need the same UX for consistency)

@reusr1 I had this in mind and thought about taking object-view and asset-view unification (as present here) in as issue for the future.
About the missing lines and the icons I'd appreciate screenshots to be able to get more into detail

Commit:
Merge branch 'up-develop' into pr/unify-page-and-templateview

* up-develop: (29 commits)
  New site creation now replaces references to old content path in site css. (#244)
  add tags to asset node rendering
  fixes for tags
  headline field (feature) (#108)
  fix package lock file
  Issues/149 (#239)
  Fixed missing assets folder after site creation. (#238)
  #157: made input type=range 5px smaller to avoid triggering the scrollbar
  Backport of the fixes for the Container / Multi Components Copy
  #233: added mixin for created date to asset
  Removed the miliseconds from the ECMA date format since JS appears unable to digest them.
  #195: fix for datetime picker popping up modal when in preview mode
  #192 - By default new Assets will be replaced if they already exists
  #231 - Fix for Images with special character during Asset Upload
  #230 - Images with spaces / special character is failing the preview
  #165 - Enhanced the application of given properties to apply child nodes as well
  Excluded S3 tests from IT tests
  issues/105: Fix for "NoSuchNodeTypeException: per:Replication" (#225)
  #165 - Fixed an issue where properties are not fully copied by moving them after component node can be copied
  Feature/resize observer polyfill (#222)
  ...

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/pageview/template.vue

reusr1 commented:
> > @Felix-Puetz it seems your change alters the style a bit and is not in line with the objects and assets preview (borders around the actions missing due to changing from li to a tag). I am also not sure about the icon on the left (it may be a good idea, however, then the assets and objects view need the same UX for consistency)
> 
> @reusr1 I had this in mind and thought about taking object-view and asset-view unification (as present here) in as issue for the future.
> About the missing lines and the icons I'd appreciate screenshots to be able to get more into detail

Felix,
 
![image](https://user-images.githubusercontent.com/779992/66315426-a3c2c800-e8ca-11e9-9564-4c9aa9afe81c.png) has no lines between the icons vs what you see in assets: 
![image](https://user-images.githubusercontent.com/779992/66315532-d7055700-e8ca-11e9-9c3b-0381e028355c.png)

flx-sta commented:
Oh I see.

And i guess with "the item on the left" you are refering to the "gear" icon.

Is there a problem to adapt this layout? Adding the lines should not be too big of a deal

reusr1 commented:
> Oh I see.
> 
> And i guess with "the item on the left" you are refering to the "gear" icon.

yes. the lines are not rendering because the items are not <LI> items anymore

> 
> Is there a problem to adapt this layout? Adding the lines should not be too big of a deal

it's ok - the icon seems a bit massive and why is it not in the corner (I am not a designer :-)?

flx-sta commented:
I've created a task for these changes in our System 👍

Commit:
METDEV-1908 merge asset-/objectview into explorerpreviewcontent

Commit:
METDEV-1908 fix some js edge cases errors

Commit:
METDEV-1908 replace v-else-if with v-if

Commit:
METDEV-1908 replace all nav-links with new component

- explorepreviewnavitem added
- clean up code
- add new constants for better code quality

Commit:
METDEV-1908 simplified layout. fix some errors due to recent changes

Commit:
METDEV-1908 remove headwire copyright from assetview

Commit:
METDEV-1908 add new const "NodeType"

Commit:
METDEV-1908 check show property on node type "asset". add isImage()

Commit:
METDEV-1908 fix mime type for tiff ("timage/tiff" -> "image/tiff")

Commit:
METDEV-1908 add constant MimeType including all image mime types

Commit:
METDEV-1908 make use of constant MimeType

Commit:
METDEV-1908 set new rename icon

Commit:
METDEV-1908 remove (now) obsolete "isSvg" property

Commit:
METDEV-1908 remove (now) obsolete "isSvg" property

Commit:
METDEV-1908 remove (now) obsolete "isSvg" property

Commit:
METDEV-1908 return obj.show instead of obj for asset

- CODE DOES INCLUDE BUGS AND IS NOT WORKING!

Commit:
METDEV-1908 add const Icon.LIST

Commit:
METDEV-1908 merge asset-view

- add condition for tab-menu since it's not needed for assets
- add new nav-item "references"
- move onCancel() closer to onEdit()

Commit:
METDEV-1908 apply es6 function format where not yet applied

Commit:
METDEV-1908 merge asset-view

- change scope of v-if for hasOgTags
- move nav item for references to left-nav and fix v-if condition
- overhauled layout to be able to provide a tab for the references
- add new Tab.References
- add groups for nodeTypes
- replace old computed edit with data edit and add a watch to transfer value to state
- add new const Icon.LIST and Icon.CHECK

Commit:
METDEV-1908 merge asset-view

- add -info-view to info/og-tags tab
- isImage is now computed and take no parameter anymore
- replace style tags in info-view with classes
- make info view conditional since page/template don't need it
- add hasInfoView()

Commit:
METDEV-1908 add lines on left of nav item for right nav

Commit:
METDEV-1908 remove info-view from edit mode. remove debugging code

Commit:
METDEV-1908 add new const NodeType.OBJECT

Commit:
METDEV-1908 add border left for first-child of right nav

Commit:
METDEV-1908 merge object-view

- add new node type group "select" and make use of it
- add exceptional behaviour of objects on getting schema

Commit:
METDEV-1908 merge object-view

- add objectNode method
- add ignore for "allowOperations" for NodeType.OBJECT

Commit:
METDEV-1908 merge objectview into explorerpreviewcontent

Commit:
Merge remote-tracking branch 'upstream/develop' into pr/unify-page-and-templateview

# Conflicts:
#	admin-base/ui.apps/src/main/js/constants.js

flx-sta committed:
Update template.vue

reusr1 reviewed (dismissed):
it seems this PR does not work correctly when merged with develop - all page/template/assets/object properties are missing. May be realted to the metadata tab as that is not in develop

Commit:
remove og-tags from pr. replace new rename icon with old one (slot)

Commit:
Merge remote-tracking branch 'origin/pr/unify-page-and-templateview' into pr/unify-page-and-templateview

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewnavitem/template.vue

flx-sta reviewed (commented):
👍

Commit:
Merge remote-tracking branch 'upstream/develop' into pr/unify-page-and-templateview

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/templateview/template.vue

Commit:
Merge remote-tracking branch 'origin/pr/meta-tags' into pr/unify-page-and-templateview

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/assetview/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/objectview/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/pageview/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/templateview/template.vue
#	platform/commons/src/main/java/com/peregrine/commons/util/PerConstants.java

Commit:
Revert "remove og-tags from pr. replace new rename icon with old one (slot)"

This reverts commit 7604798d

flx-sta commented:
@reusr1 finish merging pr/meta-tags #198 into this one

reusr1 committed:
fix for v-else without v-if

reusr1 commented on pagerenderer/vue/ui.apps/src/main/content/jcr_root/apps/pagerender/vue/structure/page/og_tag_dialog.json:
```diff
@@ -0,0 +1,68 @@
+      "browserRoot": "/content/assets",
+      "browserType": "image",
+      "label": "og:image",
+      "readonly": true,
```
@Felix-Puetz any reason why the og:image property would be read only? seems to not make any sense

flx-sta replied:
We used to do that to force the user into using the pathbrowser. But in the end we had to undo this because its nonsense. So we can remove that

flx-sta committed:
Update og_tag_dialog.json

reusr1 reviewed (dismissed):
looks like the image preview is now splitting the image and the data into a 50/50 split - does not seem that nice as we loose a lot of screen real estate this way

Commit:
METDEV-1721 (fix asset preview taking up 50% of the screen)

(not part of the ticket but too few to have a ticket for it)

flx-sta reviewed (commented):
👍

flx-sta commented:
~~I don't quite understand why its having conflicts. It's telling me "up to date" when i try to merge develop~~

Nevermind. It was not properly fetched

Commit:
Merge branch 'upstream-develop' into pr/unify-page-and-templateview

# Conflicts:
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/editor/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/objectview/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/pageview/template.vue
#	admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/templateview/template.vue
#	admin-base/ui.apps/src/main/js/constants.js

flx-sta commented:
Please make sure these changes will not break anything due to some og-tag stuff being merged by accident

reusr1 committed:
move margin to padding

otherwise a scrollbar appears in the asset preview

reusr1 committed:
fix for non image assets to be displayed correctly
